core/types/printing.cc

```diff
@@ -154,8 +154,24 @@ string AndType::toStringWithTabs(const GlobalState &gs, int tabs) const {
                        rightBrace ? ")" : "");
 }
 
+optional<string> showAnds(const GlobalState &, TypePtr, TypePtr);
+
+optional<string> showAndElem(const GlobalState &gs, TypePtr ty) {
+    if (auto andType = cast_type<AndType>(ty.get())) {
+        return showAnds(gs, andType->left, andType->right);
+    }
+    return make_optional(ty->show(gs));
+}
+
+optional<string> showAnds(const GlobalState &gs, TypePtr left, TypePtr right) {
+    auto leftStr = showAndElem(gs, left);
+    auto rightStr = showAndElem(gs, right);
+    return make_optional(fmt::format("{}, {}", *leftStr, *rightStr));
+}
```

Comment:
I think that neither of `showAndElem` or `showAnds` need to return `std::optional` anymore (there's no construction of `std::nullopt` anywhere now).

Reply:
🤦🏽 fixed!